libs/membership/api/src/lib/subscription/subscription.service.ts

```diff
@@ -75,9 +75,25 @@ export class SubscriptionService {
         deactivation: true,
         user: true,
         paymentMethod: true,
-        memberPlan: true
+        memberPlan: true,
+        invoices: {
+          where: {
+            paidAt: null,
+            canceledAt: null
+          }
+        }
       }
     })
+
+    // Ensure that no invoice gets created if there is a open invoice
+    const subscriptions = []
+    for (const subscription of allSubscriptions) {
+      if (subscription.invoices.length === 0) {
+        subscriptions.push(subscription)
+      }
+    }
+
+    return subscriptions
```

Comment:
Make it more readable
`return allSubscriptions.filter(subscription => !subscription.invoices.length)`